pkg/s3select/output.go

```diff
@@ -0,0 +1,623 @@
+func writeHeaderSize(headerLength int) []byte {
+	totalHeaderLen := make([]byte, 4)
+	totalLen := uint32(headerLength)
+	binary.BigEndian.PutUint32(totalHeaderLen, totalLen)
+	return totalHeaderLen
+}
+
+//  writeCRC writes the CRC for both the prelude and and the end of the protocol.
+func writeCRC(myBuffer []byte) []byte {
+	// Calculate the  CRC here:
+	myCRC := make([]byte, 4)
+	cksum := crc32.ChecksumIEEE(myBuffer)
+	binary.BigEndian.PutUint32(myCRC, cksum)
+	return myCRC
+}
+
+// writePayload writes the Payload for those protocols which the Payload is
+// necessary.
+func writePayload(myPayload string) []byte {
+	convertedPayload := []byte(myPayload)
+	payloadStore := make([]byte, len(convertedPayload))
+	copy(payloadStore[0:], myPayload)
+	return payloadStore
+}
+func writeRecordHeader() []byte {
+	var currentMessage = &bytes.Buffer{}
+
+	headerNameBytes := make([]byte, 1)
+	headerNameBytes[0] = 11
+	currentMessage.Write(headerNameBytes)
+
+	headerName := make([]byte, 11)
+	copy(headerName[0:], ":event-type")
+	currentMessage.Write(headerName)
+
+	headerType := make([]byte, 1)
+	headerType[0] = 7
+	currentMessage.Write(headerType)
+
+	headerNameSize := make([]byte, 2)
+	binary.BigEndian.PutUint16(headerNameSize, 7)
+	currentMessage.Write(headerNameSize)
+
+	headerValue := make([]byte, 7)
+	copy(headerValue[0:], "Records")
+	currentMessage.Write(headerValue)
+
+	// Creation of the Header for Content-Type 	// 13 -content-type -7 -24
+	// "application/octet-stream"
+
+	contentHeaderNameBytes := make([]byte, 1)
+	contentHeaderNameBytes[0] = 13
+	currentMessage.Write(contentHeaderNameBytes)
+
+	contentHeaderName := make([]byte, 13)
+	copy(contentHeaderName[0:], ":content-type")
+	currentMessage.Write(contentHeaderName)
+
+	contentHeaderType := make([]byte, 1)
+	contentHeaderType[0] = 7
+	currentMessage.Write(contentHeaderType)
+
+	contentHeaderNameSize := make([]byte, 2)
+	binary.BigEndian.PutUint16(contentHeaderNameSize, 24)
+	currentMessage.Write(contentHeaderNameSize)
+
+	contentheaderValue := make([]byte, 24)
+	copy(contentheaderValue[0:], "application/octet-stream")
+	currentMessage.Write(contentheaderValue)
+
+	// Creation of the Header for message-type 13 -message-type -7 5 "event"
+
+	messageHeaderNameBytes := make([]byte, 1)
+	messageHeaderNameBytes[0] = 13
+	currentMessage.Write(messageHeaderNameBytes)
+
+	messageHeaderName := make([]byte, 13)
+	copy(messageHeaderName[0:], ":message-type")
+	currentMessage.Write(messageHeaderName)
+
+	messageHeaderType := make([]byte, 1)
+	messageHeaderType[0] = 7
+	currentMessage.Write(messageHeaderType)
+
+	messageHeaderNameSize := make([]byte, 2)
+	binary.BigEndian.PutUint16(messageHeaderNameSize, 5)
+	currentMessage.Write(messageHeaderNameSize)
+
+	messageHeaderValue := make([]byte, 5)
+	copy(messageHeaderValue[0:], "event")
+	currentMessage.Write(messageHeaderValue)
+	return currentMessage.Bytes()
+}
+func writeEndHeader() []byte {
```

Comment:
For each of these functions, include a link to the specification so it is possible to check if the implementation is correct.